cdap-metadata-ext-spanner/src/main/java/io/cdap/cdap/metadata/spanner/SpannerMetadataStorage.java

```diff
@@ -25,36 +31,263 @@
+   * <pre>
+   * metadata_id  namespace   entity_type   props_name    props_scope   props_value
+   * D_1          default     type_1       props_name_1   USER          value_1
+   * D_1          default     type_1       props_name_2   SYSTEM        value_2
+   * D_2          system      type_2       props_name_1   SYSTEM        value_3
+   * D_2          system      type_2       props_name_2   SYSTEM        value_4
+   * D_3          default     type_3       props_name_2   SYSTEM        value_5
+   * D_3          system      type_3       props_name_2   SYSTEM        value_6
+   * </pre>
+   */
+  private void createMetadataSchema() throws IOException {
+    List<String> ddlStatements = new ArrayList<>();
+    ddlStatements.add(metadataTableDdlStatement());
+    ddlStatements.add(metadataPropsTableDdlStatement());
+    ddlStatements.addAll(getAllSearchIndexDdlStatements());
+    executeDdlStatements(ddlStatements);
+  }
+
+  /**
+   * <p>Key features of the schema:
+   *     <ul>
+   *         <li>**`metadata_id`:** An Unique which is used to identify the various
+   *         components of the pipeline metadata.
+   *         </li>
+   *         <li>**`namespace`:** It contains the namespace associated with a given entity.
+   *         </li>
+   *         <li>**`entity_type`:** It contains the type of entity.
+   *         </li>
+   *         <li>**`name`:** Name of the entity.
+   *         </li>
+   *         <li>**`created`:** creation time of the entity.</li>
+   *         <li>**`user`:** user scope related properties and tags</li>
+   *         <li>**`system`:**system scope related properties and tags </li>
+   *         <li>**`metadata_column`:** contains all the metadata related </li>
+   *         <li>user_tokens, system_tokens and text_tokens are the Tokenlists of User,
+   *         System on which we create search indexes and use them for user, system
+   *         and null scope searches respectively.</li>
+   *     </ul>
+   * </p>
+   * TODO(CDAP-21174): Add the TTl policies for cleanups.
+   */
+  private String metadataTableDdlStatement() {
+    return String.format(
+      "CREATE TABLE IF NOT EXISTS %s (" +
+        "%s STRING(MAX) NOT NULL," +  // metadata_id
+        "%s STRING(MAX) NOT NULL," + // namespace
+        "%s STRING(MAX) NOT NULL," + // entity_type
+        "%s STRING(MAX) NOT NULL," + // name
+        "%s INT64," + // creation-time
+        "%s STRING(MAX)," + // user
+        "%s STRING(MAX)," + // system
+        "%s JSON," + // metadata_column
+        "%s INT64 NOT NULL," +
+        "user_tokens TOKENLIST AS " +
+        "(TOKENIZE_SUBSTRING(USER, ngram_size_min=>1, ngram_size_max=>3)) HIDDEN," +
+        "system_tokens TOKENLIST AS " +
+        "(TOKENIZE_SUBSTRING(SYSTEM, ngram_size_min=>1, ngram_size_max=>3)) HIDDEN," +
+        "text_tokens TOKENLIST AS " +
+        "(TOKENLIST_CONCAT([User_Tokens, System_Tokens])) HIDDEN," +
+        ")PRIMARY KEY (%s) ", // primary key
```

Comment:
`// primary key` comment is obvious as it is next to the code: `")PRIMARY KEY (%s) "`

It would be useful if the primary key column name is mentioned.